code/API_definitions/device-swap.yaml

```diff
@@ -0,0 +1,399 @@
+paths: 
+  /retrieve-date:
+    post:
+      tags:
+        - Retrieve Device Swap Date
+      description: Get timestamp of last device swap for a mobile user account provided with phone number.
+      operationId: retrieveDeviceSwapDate
+      security:
+        - openId:
+          - device-swap
+        - openId:
+          - device-swap:retrieve-date
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Create a device swap date request for a phone number.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateDeviceSwapDate"
+        required: true
+      responses:
+        "200":
+          description: Contains information about Device swap change
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DeviceSwapInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+        "504":
+          $ref: "#/components/responses/Generic504"
+  /check:
+    post:
+      tags:
+        - Check Device Swap
+      description: Check if device swap has been performed during a past period
+      operationId: checkDeviceSwap
+      security:
+        - openId:
+          - device-swap
+        - openId:
+          - device-swap:check
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Create a check device swap request for a phone number.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateCheckDeviceSwap"
+        required: true
+      responses:
+        "200":
+          description: Returns whether a device swap has been performed during a past period
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CheckDeviceSwapInfo"
+        "400":
+          $ref: "#/components/responses/Generic400Check"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "429":
+          $ref: "#/components/responses/Generic429"
```

Comment:
429 is not defined for sims swap.
I'm fine to have it here but should we add it in simswap? cc @fernandopradocabrillo